src/bp-groups/bp-groups-template.php

```diff
@@ -414,6 +414,12 @@ function bp_has_groups( $args = '' ) {
 		$parent_id = 0;
     }
 
+	$show_hidden = true;
+	if( !empty( bp_displayed_user_id() ) && !empty( bp_current_user_id() ) &&
+		bp_displayed_user_id() == bp_current_user_id()
+	) {
+		$show_hidden = false;
+	}
```

Comment:
@krunal-buddyboss We don't need to make any changes here it will always pass to `true` as per @mehulkaklotar reviewed check the SQL and see here https://github.com/buddyboss/buddyboss-platform/blob/32b924a2998e1d320cda8d679f7e4ef0762f8289/src/bp-groups/classes/class-bp-groups-group.php#L1167 as @dipskakadiya added a comment and then make the change accordingly.

Reply:
@chetansatasiya 
I have applied this changes after discussed with @dipskakadiya as some scenario was failed to show hidden groups for logged in user on profile page. After apply these changes all scenarios looks good I have also checked and verified. Thanks.
+ @mehulkaklotar